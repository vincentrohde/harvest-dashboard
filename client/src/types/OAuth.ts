export type oAuthType = {
    access_token: string;
    account_id: string;
    expiration_date: string;
    refresh_token: string;
};
