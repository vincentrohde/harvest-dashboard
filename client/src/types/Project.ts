import { categoriesType, CategoryInterface } from './Category';

export type projectsType = categoriesType;

export interface ProjectInterface extends CategoryInterface {}

export type projectsByHours = ProjectByHours[];

export interface ProjectByHours {
    project: string;
    hours: number;
}
