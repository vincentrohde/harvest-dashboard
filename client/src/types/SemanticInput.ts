export type onChangeHandler = (_event: any, { name, value }:
    { name: string; value: string }) => void;

export interface Label {
    children: string;
    htmlFor: string;
}

export interface SemanticInputBasic {
    onChange: onChangeHandler;
}

export interface SemanticInputProps extends SemanticInputBasic {
    label: Label;
    name: string;
    placeholder: string;
    value: string | number;
}
