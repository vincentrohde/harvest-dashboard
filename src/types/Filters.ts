export interface FiltersInterface {
    dateRange: string[] | 1;
}
