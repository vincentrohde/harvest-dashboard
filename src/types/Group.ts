export type groupsType = GroupInterface[];

export interface GroupInterface {
    id: number;
    name: string;
}