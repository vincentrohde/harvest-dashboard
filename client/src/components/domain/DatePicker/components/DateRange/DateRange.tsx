// Libs
import React from 'react';
import { DatesRangeInput } from 'semantic-ui-calendar-react';

// Types
import { DateRangeProps } from './DateRange.types';

const DateRange = ({ onChange, value }: DateRangeProps) => <DatesRangeInput
    className="date-picker"
    name={"dateRange"}
    label={{
        children: "Date Range",
        htmlFor: "form-select-control-task" }}
    placeholder={"From - To"}
    value={value}
    iconPosition="left"
    onChange={onChange}
/>;

export default DateRange;
