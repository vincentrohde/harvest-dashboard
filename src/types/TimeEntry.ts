import { ProjectInterface } from './Project';
import { TaskInterface } from './Task';

export type timeEntriesType = TimeEntryInterface[];

export interface ShallowTimeEntryInterface {
    hours: number | string;
    id: number;
    notes: string;
    spent_date: string;
    is_running: boolean;
    project: ProjectInterface;
    task: TaskInterface;
}

export interface TimeEntryInterface extends ShallowTimeEntryInterface {
    hours: number;
}

export interface ShallowSubmissionEntryInterface {
    notes: ShallowTimeEntryInterface['notes'];
    spent_date: ShallowTimeEntryInterface['spent_date'];
    hours: ShallowTimeEntryInterface['hours'];
    project_id: number | string;
    task_id: number | string;
}

export interface TimeEntrySubmissionInterface extends ShallowSubmissionEntryInterface {
    hours: TimeEntryInterface['hours'];
    project_id: number;
    task_id: number;
}