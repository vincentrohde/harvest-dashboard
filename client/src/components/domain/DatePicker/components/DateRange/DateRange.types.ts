// Types
import { onChangeHandler } from '@/types/SemanticInput';

export interface DateRangeProps {
    value: string;
    onChange: onChangeHandler;
}
