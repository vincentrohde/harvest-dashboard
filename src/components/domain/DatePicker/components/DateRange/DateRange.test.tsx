// Libs
import { render } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

// Components
import DateRange from './DateRange';

// Types
import { DateRangeProps } from './DateRange.types';

const mockData: DateRangeProps = {
    value: 'Test',
    // @ts-ignore
    onChange: (_event: 'any', data) => null
}

describe('<DateRange />', () => {
    it('renders', () => {
        const { asFragment } = render(
            <DateRange {...mockData} />
        );

        expect(asFragment()).toMatchSnapshot();
    });
})
