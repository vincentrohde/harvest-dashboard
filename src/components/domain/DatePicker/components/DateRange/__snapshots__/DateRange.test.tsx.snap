// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateRange /> renders 1`] = `
<DocumentFragment>
  <div
    class="field date-picker"
  >
    <label
      for="form-select-control-task"
    >
      Date Range
    </label>
    <div
      class="ui left icon input"
    >
      <i
        aria-hidden="true"
        class="calendar icon"
      />
      <input
        name="dateRange"
        placeholder="From - To"
        tabindex="0"
        type="text"
        value="Test"
      />
    </div>
  </div>
</DocumentFragment>
`;
